src/Chart/Chart.jsx

```diff
@@ -96,10 +102,31 @@ export default (ChartElement) => class ChartComponent extends Component {
         dataset.data.push(dataObject[dataset.dataProperty])
       }
     }
+    
+    data.datasets = this.populateWithColors(data.datasets)
 
     return data
   }
   
+  populateWithColors (datasets) {
+    // update the colors on the datasets.
+    const datasetColors = this.props.colorPalette(datasets.length, this.props.colorScale)
+    
+    // Set the colors of the dataset if needed by ChartElement.
+    datasetColors.datasetColors = datasetColors
+    
+    // Make a copy of each of the datasets before passing them along
+    datasets = datasets.map((dataset, index) => {
+      return {
+        ...dataset,
+        backgroundColor: dataset.backgroundColor || chroma(datasetColors[index]).alpha(0.4).css(),
```

Comment:
when will a dataset not have a `backgroundColor` property? or when will it? It seems like it will almost never have one. so we may just be able to get rid of this and go with

```javascript
datasets = datasets.map((dataset, index) => {
  const backgroundColor = chroma(datasetColors[index]).alpha(0.4).css()
  const borderColor = datasetColors[index]
  
  return {
    ...dataset,
    backgroundColor,
    borderColor
  }
})